src/test/java/space/obminyashka/items_exchange/service/RefreshTokenServiceTest.java

```diff
@@ -1,55 +1,125 @@
 class RefreshTokenServiceTest {
+    private final static String EXPECTED_USERNAME = "user";
+    private final static String EXPECTED_ACCESS_TOKEN = "access_token";
+    private final static String EXPECTED_REFRESH_TOKEN = "refresh_token";
     @Mock
     private JwtTokenService jwtTokenService;
     @Mock
     private RefreshTokenRepository refreshTokenRepository;
     @InjectMocks
     private RefreshTokenServiceImpl refreshTokenService;
 
-    @ParameterizedTest
-    @MethodSource("testUsersWithRefreshTokens")
-    void createRefreshToken_shouldCreateOrUpdateToken(User testUser) {
-        when(jwtTokenService.generateRefreshToken(anyString())).thenReturn("token_string");
+
+    @Test
+    void createRefreshToken_whenRefreshTokenIsNullOrDoesNotExistInDB_shouldCreateToken() {
+        var mockRefreshToken = new RefreshToken().setToken(null);
+        mockJwtRefreshTokenData(EXPECTED_REFRESH_TOKEN);
+
+        final var actualRefreshToken = refreshTokenService.createRefreshToken(mockRefreshToken.getToken(), EXPECTED_USERNAME);
+
+        assertAll(
+                () -> assertEquals(EXPECTED_REFRESH_TOKEN, actualRefreshToken.getToken()),
+                () -> verify(jwtTokenService).generateRefreshToken(EXPECTED_USERNAME),
+                () -> verify(jwtTokenService).generateRefreshTokenExpirationTime(),
+                () -> verify(refreshTokenRepository)
+                        .createRefreshToken(eq(EXPECTED_USERNAME), eq(EXPECTED_REFRESH_TOKEN), any())
+        );
+    }
+
+    @Test
+    void createRefreshToken_whenRefreshTokenExistsInDB_shouldUpdateToken() {
+        var expectedRefreshToken = new RefreshToken().setToken("");
+        mockJwtRefreshTokenData(expectedRefreshToken.getToken());
+
+        final var actualRefreshToken = refreshTokenService
+                .createRefreshToken(expectedRefreshToken.getToken(), EXPECTED_USERNAME);
+
+        assertAll(
+                () -> assertEquals(expectedRefreshToken.getToken(), actualRefreshToken.getToken()),
+                () -> verify(jwtTokenService).generateRefreshToken(EXPECTED_USERNAME),
+                () -> verify(jwtTokenService).generateRefreshTokenExpirationTime(),
+                () -> verify(refreshTokenRepository)
+                        .updateRefreshToken(eq(EXPECTED_USERNAME), eq(expectedRefreshToken.getToken()), any())
+        );
+    }
+
+    private void mockJwtRefreshTokenData(String refreshToken) {
+        when(jwtTokenService.generateRefreshToken(anyString())).thenReturn(refreshToken);
         when(jwtTokenService.generateRefreshTokenExpirationTime()).thenReturn(LocalDateTime.MAX);
-        when(refreshTokenRepository.save(any())).thenAnswer(invocation -> invocation.getArgument(0));
+    }
 
-        final var refreshToken = refreshTokenService.createRefreshToken(testUser);
+    @Test
+    void renewAccessTokenByRefreshToken_whenAccessTokenExpires_shouldRenewAccessToken() {
+        var existRefreshToken = createRefreshToken(1);
```

Comment:
The number '1' looks like here as a magic number
The variable name should get a better name describing its behavior